File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/PortfolioRequest.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import java.util.List;
import java.util.UUID;

import dev.jefy.connectpro.portfolio.domain.vo.PortfolioType;

/**
 * @author Jôph Yamba
 */
public record PortfolioRequest (
    UUID userId,
    PortfolioType type,
    GeneralInfoRequest generalInfo,
    ProfessionalInfoRequest professionalInfo,
    ContactInfoData contactInfo,
    LocationInfoData locationInfo,
    List<SocialLinkData> socialLinks  
) {}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/PortfolioRequest.kt
package dev.jefy.connectpro.portfolio.application.dtos

import java.util.*


/**
 * @author Jôph Yamba
 */
data class CreatePortfolioRequest(
    val userId: UUID,
    val name: String,
    val bio: String,
    val details: String?,
    val contact: ContactData,
    val location: LocationData,
    val socials: List<SocialData>
)

data class UpdatePortfolioRequest(
    val name: String,
    val bio: String,
    val details: String?
)
